### ISSUE TITLE
_EITC_MinEligAge is not an integer in TaxBrain GUI input

### ISSUE THREAD

**Issue by martinholmer** [OpenSourcePolicyCenter/PolicyBrain#795]
In closed [Tax-Calculator issue 1809](https://github.com/open-source-economics/Tax-Calculator/issues/1809), TaxBrain user @siervicul reported this problem:

I'm using the TaxBrain interface, and when changing the minimum age from the default of 25 to 21, I get the error:
```
ERROR: _EITC_MinEligAge value 21.0 is not integer for 2017
```
Even when I change it back to 25 to try to clear the error message, the error becomes:
```
ERROR: _EITC_MinEligAge value 25.0 is not integer for 2017
```

**Comment by hdoupe** [comment]
@martinholmer do you think #795 is related to open-source-economics/Tax-Calculator#1782?  Tax-Calculator may be doing a check to see if this is an integer and adding an error message before checking if it is a float.

**Comment by martinholmer** [comment]
@hdoupe said:
> Tax-Calculator may be doing a check to see if this is an integer and adding an error message before checking if it is a float.

Tax-Calculator has never allowed a float value for an integer policy parameter since we added the `integer_value` and `boolean_value` fields to each parameter record in `current_law_policy.json`, which happened in [Tax-Calculator pull request 1585](https://github.com/open-source-economics/Tax-Calculator/pull/1585) that was in Tax-Calculator 0.12.0 released on October 20, 2017.

Here is the `_EITC_MinEligAge` record in `current_law_policy.json`:
```
    "_EITC_MinEligAge": {
        "long_name": "Minimum Age for Childless EITC Eligibility",
        "description": "For a childless filling unit, at least one individual's age ...",
        "section_1": "Refundable Credits",
        "section_2": "Earned Income Tax Credit",
        "irs_ref": "Form 1040, line 66a&b, step 4, instructions.",
        "notes": "",
        "row_var": "FLPDYR",
        "row_label": ["2013"],
        "start_year": 2013,
        "cpi_inflated": false,
        "col_var": "",
        "col_label": "",
        "boolean_value": false,
        "integer_value": true,                          <==========================
        "value": [25],
        "range": {"min": 0, "max": 125},
        "out_of_range_minmsg": "",
        "out_of_range_maxmsg": "",
        "out_of_range_action": "stop"
    },
```

**Comment by hdoupe** [comment]
OK, I see.  I'm looking at this code snippet in [taxcalc/policy.py](https://github.com/open-source-economics/Tax-Calculator/blob/master/taxcalc/policy.py#L451-L458):

```
elif int_type:
    if not isinstance(pvalue[idx], int):
        msg = '{} {} value {} is not integer'
        self.reform_errors += (
            'ERROR: ' +
            msg.format(year, pname, pvalue[idx]) +
            '\n'
)
```

It seems like we could alter the condition to be:
```
elif int_type:
    if ((not isinstance(pvalue[idx], int)) or 
        (isinstance(pvalue[idx], float) and pvalue[idx] % 1 == 0.0)):
...
```

since we have:
```
In [2]: 2.5 % 1
Out[2]: 0.5

In [3]: 2.0 % 1
Out[3]: 0.0
```

@martinholmer is there a reason that I am missing for why Tax-Calculator does not accept float values that are integer-like?

**Comment by hdoupe** [comment]
The disagreement between PolicyBrain and Tax-Calculator parameter types seems to be a recurring problem.  @martinholmer I think we need an issue that lays out the different types of parameters and how PolicyBrain and Tax-Calculator should process them.  Otherwise, I think this problem will continue to pop up.

**Comment by martinholmer** [comment]
@hdoupe said:
> The disagreement between PolicyBrain and Tax-Calculator parameter types seems to be a recurring problem. @martinholmer I think we need an issue that lays out the different types of parameters and how PolicyBrain and Tax-Calculator should process them. Otherwise, I think this problem will continue to pop up.

Why doesn't TaxBrain simply use the types specified in the `current_law_policy.json` file?
Is that approach difficult to implement?

**Comment by martinholmer** [comment]
A further look at the current production version of TaxBrain shows that integer policy parameters are being shown as floats.  See, for example, this part of the GUI input page:

![screen shot 2018-01-05 at 8 12 15 am](https://user-images.githubusercontent.com/12170745/34610807-c3a7006e-f1f0-11e7-8617-3d992e7fcf01.png)

Showing the two integer parameters as floats is highly misleading to TaxBrain users because it implies that they can specify values like 24.5, which are nonsensical because our input data does not have birth dates but only integer ages.

All the integer policy parameters should be displayed on the TaxBrain GUI input page as integers, not floats.

**Comment by hdoupe** [comment]
Thanks for your input @martinholmer.  Since this is going to require some substantial changes, I think we should push this issue to the next release.

**Comment by martinholmer** [comment]
@hdoupe said:
> Since this is going to require some substantial changes, I think we should push this issue to the next release.

Sounds sensible.

**Comment by hdoupe** [comment]
Closed via #822
